### PR TITLE
Always attempt to resync when receiving a block less than LIB.

### PR DESCRIPTION
**Pull request by jgiszczak**
## Change Description
Option to allow nodeos to continue syncing with a peer that has sent blocks older than LIB due to latency.

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [x] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by nksanthosh** [comment]
Continue testing for today and Kevin can review as soon as he is back.

File: plugins/net_plugin/net_plugin.cpp [hunk 1]
```diff
@@ -2918,9 +2919,18 @@ namespace eosio {
...
+
             if( blk_num < last_sent_lib ) {
                fc_ilog( logger, "received block ${n} less than sent lib ${lib}", ("n", blk_num)("lib", last_sent_lib) );
-               close();
+               if(my_impl->p2p_always_resync) {
+                  fc_ilog( logger, "attempting to resync with peer that is sending blocks less than sent lib ${lib}", ("lib", last_sent_lib) );
+                  enqueue( (sync_request_message) {0, 0} );
+                  send_handshake();
+                  cancel_wait();
+               }
```
Comment (heifner) [inline plugins/net_plugin/net_plugin.cpp:2930]: I think we should not add this new option and just make this the behavior.

Comment (nksanthosh) [reply plugins/net_plugin/net_plugin.cpp:2930]: Please put in your review comments and you can approve when it is done.

File: plugins/net_plugin/net_plugin.cpp [hunk 2]
```diff
@@ -3971,6 +3981,7 @@ namespace eosio {
            "   _port  \tremote port number of peer\n\n"
            "   _lip   \tlocal IP address connected to peer\n\n"
            "   _lport \tlocal port number connected to peer\n\n")
+         ( "p2p-always-resync", bpo::value<bool>()->default_value(false), "Always attempt to resync even when peer sends blocks earlier than this node's last irreversible block")
```
Comment (heifner) [inline plugins/net_plugin/net_plugin.cpp:3984]: I think we should not add this new option and just make this the behavior.

**Review by heifner — CHANGES_REQUESTED** [review]

File: plugins/net_plugin/net_plugin.cpp (hunk 2 above)
Comment (jgiszczak) [reply plugins/net_plugin/net_plugin.cpp]: Option removed.

Comment (heifner) [reply plugins/net_plugin/net_plugin.cpp]: PR name/description needs to be updated.

Comment (jgiszczak) [reply plugins/net_plugin/net_plugin.cpp]: Updated.

**Review by heifner — APPROVED** [review]

**Pull request merged**